### PR TITLE
fix: Remove more general duplicated "user" block in Helm chart

### PR DESCRIPTION
**Pull request by lgw-gravitee**
fixes AM-3313, also updated with values not present in Helm chart, but in gravitee.yml: mfaVerifyAttempt and webAuthn

### CODE REVIEW

**Review by leleueri — CHANGES_REQUESTED** [review]
**TODO:** can you please add unit test to validate the final config file ?

**Comment by lgw-gravitee** [comment]
I wasnt aware there are tests for helm charts ;)
anyway, I added them and chcanged it according to your comment.

File: helm/templates/gateway/gateway-configmap.yaml
Comment (leleueri) [inline helm/templates/gateway/gateway-configmap.yaml:1]: **TODO**: looks like we have the same block on management-api configmap https://github.com/gravitee-io/gravitee-access-management/blob/master/helm/templates/api/api-configmap.yaml#L319 can you please update this one too ?

File: helm/tests/gateway/configmap_user_test.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        email: test@test.com
+        password: password
+        name: name
+        username: luk
+        registration: 1
+        resetPassword: 2
+        blockedAccount: 3
+        mfaChallenge: 4
+        mfaVerifyAttempt: 5
+        webAuthn: 6
```
Comment (leleueri) [inline helm/tests/gateway/configmap_user_test.yaml:18]: **TODO**: is it possible to test at lest one complexe entry like it should be in a valid config?
ex:
```
  email:
    policy:
      pattern: ^[a-zA-Z0-9_+-]+(?:\.[a-zA-Z0-9_+-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,15}$
```

File: helm/tests/gateway/configmap_user_test.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: ConfigMap
+      - matchRegex:
+          path: data.[gravitee.yml]
+          pattern: "user:\n( *[a-zA-Z]*: .*\n)* *email: test@test.com"
```
Comment (leleueri) [inline helm/tests/gateway/configmap_user_test.yaml:26]: **suggestion**: Looks like we can use a text block as regex. It should help for readability and indentation validation.
ex : https://github.com/gravitee-io/gravitee-api-management/blob/master/helm/tests/gateway/confiigmap_db_less_test.yaml#L31

```
      - matchRegex:
          path: data.[gravitee.yml]
          pattern: |
            email:
              policy:
                pattern: ...
```

**Review by leleueri — CHANGES_REQUESTED** [review]

File: helm/tests/gateway/configmap_user_test.yaml (hunk 2 above)
Comment (lgw-gravitee) [reply helm/tests/gateway/configmap_user_test.yaml:26]: fixed

File: helm/tests/gateway/configmap_user_test.yaml (hunk 1 above)
Comment (lgw-gravitee) [reply helm/tests/gateway/configmap_user_test.yaml:18]: added

File: helm/templates/gateway/gateway-configmap.yaml
Comment (lgw-gravitee) [reply helm/templates/gateway/gateway-configmap.yaml:1]: updated

**Review by leleueri — APPROVED** [review]

**Pull request merged**